smack-core/src/main/java/org/jivesoftware/smack/packet/Stanza.java

```diff
@@ -458,6 +468,144 @@ public ExtensionElement removeExtension(ExtensionElement extension)  {
         return removeExtension(extension.getElementName(), extension.getNamespace());
     }
 
+    /**
+     * Returns <tt>true</tt> if custom attributes management is enabled.
+     * @return <tt>true</tt> if custom attributes management is enabled.
+     */
+    public static boolean isCustomAttributesEnabled() {
+        return customAttributesEnabled;
+    }
+
+    /**
+     * Enables the possibility to add custom attributes to the stanza.
+     * <br/><b>Enabling this feature breaks the XMPP standard and may induce unexpected behaviors!
+     * <br/>If your use-case allow it, prefer using ExtensionElement instead, and if not, be sure to triple-check the result.</b>
+     */
+    public static void enableCustomAttributes() {
+        customAttributesEnabled = true;
+    }
+
+    private static void requireCustomAttributesEnabled(String methodName) {
```

Comment:
If you really want to the callers method name here, then get it via runtime mechanism instead of hardcoding it at the call site. All in all, I don't think it's really required, so just remove the methodName argument.